LZ4/src/H5Zlz4.c

```diff
@@ -143,19 +143,24 @@ static size_t H5Z_filter_lz4(unsigned int flags, size_t cd_nelmts,
             else /* do the decompression */
             {
 #if LZ4_VERSION_NUMBER > 10900
-                int compressedBytes = LZ4_decompress_safe(rpos, roBuf, blockSize, origSize);
+                int compressedBytes = LZ4_decompress_safe(rpos, roBuf, compressedBlockSize, origSize);
+                if(compressedBytes != blockSize)
+                {
+                    printf("decompressed size not the same: %d, != %d\n", compressedBytes, blockSize);
+                    goto error;
+                }
```

Comment:
```suggestion
                int decompressedBytes = LZ4_decompress_safe(rpos, roBuf, compressedBlockSize, blockSize);
                if(decompressedBytes != blockSize)
                {
                    printf("decompressed size not the same: %d, != %d\n", decompressedBytes, blockSize);
                    goto error;
                }
```

From the lz4.h file, the last argument of `LZ4_decompress_safe` https://github.com/lz4/lz4/blob/dev/lib/lz4.h:

> dstCapacity : is the size of destination buffer (which must be already allocated), presumed an upper bound of decompressed size.

So it should be `blockSize` not `origSize`.

Reply:
Right, don't know how I missed this. Changed.